### ISSUE TITLE
Bug: Parsing error: Unexpected token

### ISSUE THREAD

**Issue by micro-bean** [eslint/eslint#19194]
### Environment

Node version:20.9.0
npm version:10.1.0
Local ESLint version:9.15.0
Global ESLint version:9.15.0
Operating System:Windows 11 x64

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```
<!-- Paste your configuration here -->
import globals from 'globals'
import jsEslint from '@eslint/js'
import tsEslint from 'typescript-eslint'
import vueEslint from 'eslint-plugin-vue'

export default tsEslint.config(
    jsEslint.configs.recommended,
    tsEslint.configs.recommended,
    vueEslint.configs['flat/recommended'],
    {
        files: ['**/*.ts', '**/*.vue'],
        languageOptions: {
            sourceType: 'module',
            ecmaVersion: 'latest',
            globals: { ...globals.browser, ...globals.node, ...globals.es2021 }
        },
        rules: {}
    }
)
```
</details>

In the vue3 file <script setup lang="ts"></script>
```js
<!-- Paste your code here -->
function onMenu(url: string) {
    return ''
}
```

### What did you expect to happen?

No errors

### What actually happened?

URL parameter prompt：Parsing error: Unexpected token

### Link to Minimal Reproducible Example

none

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by 9romise** [comment]
You should use `@typescript-eslint/parser` to parse the typescript code.

**Comment by Tanujkanti4441** [comment]
Hi @micro-bean, thanks for the issue, but with the config and code you have provided, every thing seems correct as i tried to reproduce it locally.
![Image](https://github.com/user-attachments/assets/0233f567-3f38-48ea-b797-b5ad46f02097)

**Comment by micro-bean** [comment]
@9romise Thank you, it disappeared after use
@Tanujkanti4441 Thank you for your response
![Image](https://github.com/user-attachments/assets/0af21f78-1f05-49a8-81f6-316eba9824be)
